### DISCUSSION TITLE
FBX of animation

### DISCUSSION THREAD

**Discussion by luohuiiqng** [openscenegraph/OpenSceneGraph#1303] [Q&A] [answered]
When I use the following command, only the model is in the scene and the animation that comes with the model is not played. What can I do to automatically play the animation when I open the model file? Thank you!

`osgviewer.exe “xxx.fbx”`

**Comment by robertosfield** [comment, marked as answer]
osgviewer just loads and renders models, it doesn't have any code for finding and playing animations.

Try osganimationviewer as it demonstrates usage of animations.

Also please realize that the OSG project is in maintenance mode and meant for supporting legacy applications.  For new application developer you would be better served by looking at the VSG.  Details can be found in #1291.

**Reply by luohuiiqng** [reply]
Thanks you! VSG is very good, but there are not many examples of its use at present, and we plan to learn more about it once in a while.

**Reply by robertosfield** [reply]
There is already 73 examples in [vsgExamples](https://github.com/vsg-dev/vsgExamples) not quite as many as the OSG but closing in fast.  There is also the [vsgTutorial](https://vsg-dev.github.io/vsgTutorial/) online book, this about 1/3 completed but should be helpful even in it's incomplete state.

If you area learning OSG/OpenGL first then plan to learn VSG/Vulkan later you will be wasting lots of time overall.  The VSG is the successor to the OSG, it takes all the lessons from developing the OSG over nearly 2 decades and move the art of the scene graphs forward into the current generation.  It's faster and better designed than the OSG. 

I would also suggest that developers can learn how to write better software using the VSG as inspiration rather than the OSG.  The VSG is C++17, uses modern CMake and modern software idioms, it's more flexible and minimal than the OSG.  The OSG was a good example back in the 2000's but not now, the world has moved on.

**Reply by luohuiiqng** [reply]
Thanks you!
I think I need an example to integrate this into qt's quick.

**Reply by robertosfield** [reply]
I don't know of any Qt Quick integration for the OSG or VSG,  there is Qt C++ integration via 

https://github.com/vsg-dev/vsgQt
https://github.com/openscenegraph/osgQt

We have graphics expertise but for Qt you'll need to bring some of your own expertise to the table, or hire someone who has it.

**Reply by luohuiiqng** [reply]
Ok,I try it!
